functions/redirect-url-composer/1.0/index.js

```diff
@@ -0,0 +1,11 @@
+const redirectUrlComposer = async ({ url, placeholderMapping }) => {
+  placeholderMapping.map((placeholder) => {
+    url = url.replace(':' + placeholder.key, placeholder.value);
+  });
+
+  return {
+    output: url,
+  };
```

Comment:
You could write this as
```javascript
let output = url;

placeholderMapping.forEach(({ key, value }) => {
  output = output.replace(`:${key}`, value);
});

return {
  output,
};
```

`Array.map` is used to created a new array, see: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map

Reply:
Thanks! :)